src/builtins.c

```diff
@@ -1591,6 +1591,14 @@ JL_CALLABLE(jl_f_donotdelete)
     return jl_nothing;
 }
 
+JL_CALLABLE(jl_f__add_finalizer)
+{
+    JL_NARGS(_add_finalizer, 2, 4);
```

Comment:
What are arguments 3 and 4 supposed to be?

Reply:
The inlining pass does legality analysis and stashes the inlining target in the intrinsic. It's not the prettiest design, but I think it works well, because it keeps the inlining pass more contained.

Reply:
Since there is already a FinalizerInfo node attached, could we add the extra fields (MethodInstance and do-inline) to that does the same without making the IR weird?

Reply:
No, we don't preserve the info in the code cache and it needs to survive in case it gets inlined into a larger function later.